src/services/gemini.py

```diff
@@ -0,0 +1,73 @@
+
+import google.generativeai as genai
+import asyncio, os
+import logging
+import time
+from ratelimit import limits, sleep_and_retry
+
+# Setup logging
+logging.basicConfig(level=logging.ERROR, format="%(asctime)s, %(levelname)s: %(message)s")
+
+class GeminiPro:
+    def __init__(self):
+        # Load API key
+        api_key = os.environ.get("GEMINI_API_KEY")
+        if not api_key:
+            logging.error("GEMINI_API_KEY is missing from envrionment variables.")
+            raise ValueError("GEMINI_API_KEY is undefined.")
+        genai.configure(api_key=api_key)
+
+        # Load Gemini-1.5-Pro model
+        try:
+            self.model = genai.GenerativeModel("gemini-1.5-pro")
```

Comment:
This is fine since I didn't specify in the acceptance criteria, but note that there are other important parameters in GenerativeModel(). Here is an example:
`
genai.GenerativeModel(
            model_name=model_name,
            generation_config=generation_config,
            safety_settings=safety_settings
        )
`
For example, here generation_config specifies important attributes such as temperature, top_p, max_tokens, etc., which controls the creativity and token used for each api request. Will recommend you to research on what those attributes do and add a config setting somewhere.

Reply:
Model generation config specified for temperature, top_p, top_k, and max_output_tokens. Safety settings for hate speech and harassment specified as well.